File: Assets/Bundles/CoreClient/Resources/TS/Misc/TextureReloader.lua
-- Compiled with unity-ts v2.1.0-75
local Game = require("Shared/TS/Game").Game
local Keyboard = require("Shared/TS/UserInput/init").Keyboard
local keyboard = Keyboard.new()
keyboard.KeyDown:Connect(function(event)
	if event.Key == 102 then
		local _result = GameObject:Find("VoxelWorld")
		if _result ~= nil then
			_result = _result:GetComponent("VoxelWorld")
		end
		local voxelWorld = _result
		if not voxelWorld then
			Game.LocalPlayer:SendMessage("VoxelWorld not found.")
			return nil
		end
		voxelWorld:ReloadTextureAtlas()
		Game.LocalPlayer:SendMessage("Reloaded textures!")
	end
end)
return nil
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/CoreClient/Misc/TextureReloader.ts
import { Game } from "Shared/Game";
import { Keyboard } from "Shared/UserInput";

const keyboard = new Keyboard();
keyboard.KeyDown.Connect((event) => {
	if (event.Key === Key.F9) {
		const voxelWorld = GameObject.Find("VoxelWorld")?.GetComponent<VoxelWorld>();
		if (!voxelWorld) {
			Game.LocalPlayer.SendMessage("VoxelWorld not found.");
			return;
		}
		voxelWorld.ReloadTextureAtlas();
		Game.LocalPlayer.SendMessage("Reloaded textures!");
	}
});
